compiler/hash-ast/src/tree.rs

```diff
@@ -995,7 +1000,13 @@ impl AstVisitor for AstTreeGenerator {
         &self,
         node: ast::AstNodeRef<ast::TuplePat>,
     ) -> Result<Self::TuplePatRet, Self::Error> {
-        let walk::TuplePat { fields } = walk::walk_tuple_pat(self, node)?;
+        let walk::TuplePat { mut fields, spread } = walk::walk_tuple_pat(self, node)?;
+
+        // @@Note: this does not respect the order of the fields in the pattern
```

Comment:
What does this mean?

Reply:
it appends the item to the end of the fields, not necessarily where the spread pattern is actually specified.

Reply:
I see, though wouldn't it be better if it did?

Reply:
Something like `fields.insert(node.spread.unwrap().index, TreeNode::branch("spread", vec![spread]))`